dojo-dungeon/src/starter/DojoStarter.java

```diff
@@ -47,88 +91,106 @@ public static void main(String[] args) {
-    LevelRoom levelRoom8 = new LevelRoom(graph);
-    LevelRoom levelRoom9 = new LevelRoom(graph);
-
-    // level 4 has 3 rooms
-    LevelRoom levelRoom10 = new LevelRoom(graph);
-    LevelRoom levelRoom11 = new LevelRoom(graph);
-    LevelRoom levelRoom12 = new LevelRoom(graph);
-
-    // connect the rooms, this is needed to build the rooms in next steps
-    connectBidirectional(levelRoom1, levelRoom2);
-    connectBidirectional(levelRoom2, levelRoom3);
-    connectBidirectional(levelRoom3, levelRoom4);
-    connectBidirectional(levelRoom4, levelRoom5);
-    connectBidirectional(levelRoom5, levelRoom6);
-    connectBidirectional(levelRoom6, levelRoom7);
-    connectBidirectional(levelRoom7, levelRoom8);
-    connectBidirectional(levelRoom8, levelRoom9);
-    connectBidirectional(levelRoom9, levelRoom10);
-    connectBidirectional(levelRoom10, levelRoom11);
-    connectBidirectional(levelRoom11, levelRoom12);
-
-    // build the rooms
+    // Create a customised level comprising rooms
+    LevelRoomLevel[] allLevels = getLevelRoomLevels();
+
+    // Connect the rooms, this is needed to build the rooms in next steps
+    LevelRoom previousLevelRoom = null;
+    for (LevelRoomLevel level : allLevels) {
+      for (LevelRoom levelRoom : level.getLevelRooms()) {
+        if (previousLevelRoom != null) {
+          connectBidirectional(previousLevelRoom, levelRoom);
+        }
+        previousLevelRoom = levelRoom;
+      }
+    }
+
+    // Build the rooms (reverse order)
+    Deque<Room> rooms = new ArrayDeque<>();
     RoomGenerator gen = new RoomGenerator();
-    Room room12 = buildRoom_L4_R3_Monster_Implement_2(levelRoom12, gen, null);
-    Room room11 = buildRoom_L4_R2_Fragen_Pattern(levelRoom11, gen, room12);
-    Room room10 = buildRoom_L4_R1_Monster_3(levelRoom10, gen, room11);
-    Room room9 = buildRoom_L3_R3_Fragen_RegExes(levelRoom9, gen, room10);
-    Room room8 = buildRoom_L3_R2_Fehler_Quader(levelRoom8, gen, room9);
-    Room room7 = buildRoom_L3_R1_Fragen_Schriftrollen(levelRoom7, gen, room8);
-    Room room6 = buildRoom_L2_R3_Fehler_Refactoring(levelRoom6, gen, room7);
-    Room room5 = buildRoom_L2_R2_Monster_Implement_1(levelRoom5, gen, room6);
-    Room room4 = buildRoom_L2_R1_Monster_2(levelRoom4, gen, room5);
-    Room room3 = buildRoom_L1_R3_Fragen_Lambda(levelRoom3, gen, room4);
-    Room room2 = buildRoom_L1_R2_Fehler_Syntax(levelRoom2, gen, room3);
-    Room room1 = buildRoom_L1_R1_Monster_1(levelRoom1, gen, room2);
-
-    room1.configDoors();
-    room2.configDoors();
-    room3.configDoors();
-    room4.configDoors();
-    room5.configDoors();
-    room6.configDoors();
-    room7.configDoors();
-    room8.configDoors();
-    room9.configDoors();
-    room10.configDoors();
-    room11.configDoors();
-    room12.configDoors();
+    Room nextRoom = null;
+    for (int i = allLevels.length - 1; i >= 0; i--) {
+      LevelRoomLevel level = allLevels[i];
+      for (int j = level.getLevelRooms().length - 1; j >= 0; j--) {
```

Comment:
Eine **lesbare** foreach-Variante geht hier nicht, weil?

Also sowas wie
```
    for (LevelRoomLevel level : allLevels) {
      for (LevelRoom levelRoom : level.getLevelRooms()) {
```

Reply:
> Eine **lesbare** foreach-Variante geht hier nicht, weil?

weil wir das in umgekehrter Reihenfolge aufbauen müssen (siehe den Kommentar darüber) und es gibt keine rückwärts laufenden for-each-Schleifen

Reply:
Bzw., nur mit so einem "Monster"-Hilfskonstrukt: https://stackoverflow.com/a/1098153

Reply:
Verstehe ich grad nicht. Du hast doch Listen? Dafür gibt es sowas wie `Collections.reverse(mylist);` ...

Dein Link sagt: Ok, das Default-Reverse ist teuer (O(n)), und wenn man diese Kosten vermeiden möchte, dann soll man das über einen selbstgestrickten Iterator machen. Aber das ist doch hier gar nicht notwendig?! Zum einen ist uns $n$ sehr klein (4?), und zum anderen passiert das alles genau 1x beim Start.

Reply:
ok, das `Collections.reverse(mylist);` dreht die liste einmal in-place um. das kostet nicht viel. wir brauchen "danach" die gleiche liste aber wieder in ihrer ursprünglichen reihenfolge. das heißt, wir brauchen zwei reverse()-aufrufe. das wollte ich der lesbarkeit wegen eigentlich vermeiden.

Reply:
das klingt irgendwie nicht richtig. aber du wirst schon wissen, was du tust :)